### DISCUSSION TITLE
Allure report was skipped because there is no results directories found

### DISCUSSION THREAD

**Discussion by StanislavSurmenev** [allure-framework/allure2#2207] [Questions & Support]
Добрый день!
Собрал тест на java с использованием maven junit5 allure.
На локальной машине тест выполняется удачно и алюр-отчет успешно формируется.
Но при запуске теста на сервере gitlab ci сборка проекта и тест проходят успешно, но при этом алюр может найти allure-results и в результате отчет не формируется
Использую следующий POM

`
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>com.example.junit5</groupId>
  <artifactId>junit5-java-maven</artifactId>
  <version>1.0-SNAPSHOT</version>

  <properties>
    <maven.compiler.release>18</maven.compiler.release>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

    <allure.version>2.24.0 </allure.version>
    <aspectj.version>1.9.20.1</aspectj.version>
  </properties>

  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.qameta.allure</groupId>
        <artifactId>allure-bom</artifactId>
        <version>2.17.2</version>

        <type>pom</type>
        <scope>import</scope>
      </dependency>
      <dependency>
        <groupId>org.junit</groupId>
        <artifactId>junit-bom</artifactId>
        <version> 5.10.1 </version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>

  <dependencies>
    <dependency>
      <groupId>io.qameta.allure</groupId>
      <artifactId>allure-junit5</artifactId>
      <scope>test</scope>
    </dependency>

    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-simple</artifactId>
      <version> 2.0.9 </version>
      <scope>test</scope>
    </dependency>

    <dependency>
      <groupId>org.seleniumhq.selenium</groupId>
      <artifactId>selenium-java</artifactId>
      <version> 4.11.0 </version>
    </dependency>

    <dependency>
      <groupId>io.github.bonigarcia</groupId>
      <artifactId>webdrivermanager</artifactId>
      <version> 5.5.3 </version>
      <scope>compile</scope>
    </dependency>

    <dependency>
      <groupId>ru.yandex.qatools.allure</groupId>
      <artifactId>allure-junit-adaptor</artifactId>
      <version>1.5.4</version>
    </dependency>

  </dependencies>

  <build>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-surefire-plugin</artifactId>
        <version> 3.0.0 </version>
        <configuration>
          <systemPropertyVariables>
            <allure.results.directory>${project.build.directory}\allure-results</allure.results.directory>
            <allure.report.directory>${project.build.directory}\allure-report</allure.report.directory>
          </systemPropertyVariables>
          <testFailureIgnore>false</testFailureIgnore>
          <argLine>
            -Dfile.encoding=${project.build.sourceEncoding}
            -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
          </argLine>

          <properties>
          <property>
            <name>listener</name>
            <value>ru.yandex.qatools.allure.junit.AllureRunListener</value>
          </property>
          </properties>

        </configuration>
        <dependencies>
          <dependency>
            <groupId>org.aspectj</groupId>
            <artifactId>aspectjweaver</artifactId>
            <version>1.9.20.1</version>
          </dependency>
        </dependencies>
      </plugin>

      <plugin>
        <groupId>io.qameta.allure</groupId>
        <artifactId>allure-maven</artifactId>
        <version> 2.12.0 </version>
        <configuration>
          <reportVersion>${allure.version}</reportVersion>
          <resultsDirectory>${project.build.directory}/allure-results</resultsDirectory>
          <reportDirectory>${project.build.directory}/allure-report</reportDirectory>
        </configuration>
      </plugin>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <version> 3.11.0 </version>
        <configuration>
          <encoding>${project.build.sourceEncoding}</encoding>
          <release>18</release>
        </configuration>
      </plugin>
    </plugins>
  </build>
</project>
`

получаю такие логи

`[INFO] Generate Allure report (report) with version 2.24.0
[INFO] Generate Allure report to /builds/devops/selenium/target/allure-report
[ERROR] Directory /builds/devops/selenium/target/allure-results not found.
[WARNING] Allure report was skipped because there is no results directories found.
[INFO] 
[INFO] --- allure:2.12.0:serve (default-cli) @ junit5-java-maven ---
[INFO] Allure installation directory /builds/devops/selenium/.allure
[INFO] Try to finding out allure 2.24.0
[INFO] Generate Allure report (serve) with version 2.24.0
[INFO] Generate Allure report to /builds/devops/selenium/target/allure-report
[ERROR] Directory /builds/devops/selenium/target/allure-results not found.
[WARNING] Allure report was skipped because there is no results directories found.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:19 min
[INFO] Finished at: 2023-11-16T10:54:20Z
[INFO] ------------------------------------------------------------------------
[WARNING] 
[WARNING] Plugin validation issues were detected in 2 plugin(s)
[WARNING] 
[WARNING]  * io.qameta.allure:allure-maven:2.12.0
[WARNING]  * org.apache.maven.plugins:maven-resources-plugin:3.3.0
[WARNING] 
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING] 
Uploading artifacts for successful job
00:01
Uploading artifacts...
target/surefire-reports/*: found 5 matching artifact files and directories 
Uploading artifacts as "archive" to coordinator... 201 Created  id=12347 responseStatus=201 Created token=64_YpWmw
Cleaning up project directory and file based variables
00:00
Job succeeded
Job succeeded
`

yml Файл

`image: markhobson/maven-chrome:jdk-20

stages:
  - test

variables:
  MAVEN_OPTS: "-Dmaven.repo.local=.m2/repository"

test:
  stage: test
  allow_failure: true

  tags:
    - docker-test-runner

# Run the tests  
  script:
    - wget https://s-ncredit.neocode.kz/
    - mvn $MAVEN_OPTS clean package
    - mvn test -Dtest=TestAllPozitive
 
# Store artifacts
  artifacts:
    when: always
    name: "report"
    paths:
    - target/surefire-reports/*
    expire_in: 1 h
`

**Comment by virennitkkr** [comment]
I scanned the configuration, i found you have missed two things.

1. **Add listner in testng.xml file.**
 <listeners>
        <listener class-name="io.qameta.allure.testng.AllureTestNg"/>
    </listeners>
 2. **Add dependency in pom.xml file**
 <dependency>
          <groupId>io.qameta.allure</groupId>
          <artifactId>allure-testng</artifactId>
          <version>2.13.9</version>
          <scope>test</scope>
      </dependency>
**Run test cases:** 
mvn clean test
mvn allure:serve
